__tests__/integration/utxo-filter.test.js

```diff
@@ -0,0 +1,464 @@
+    const tkaUtxos = utxoResponse.body;
+
+    // TODO: This is returning the equivalent of "bigger_or_equal_than". Is this correct?
+    expect(tkaUtxos.total_amount_available).toBe(1000 - 10 - 20);
+    expect(tkaUtxos.total_utxos_available).toBe(4);
+    expect(tkaUtxos.utxos).toHaveProperty('length', 4);
+    expect(tkaUtxos.utxos[0]).toHaveProperty('tx_id', transactions.tx30.hash);
+    expect(tkaUtxos.utxos[1]).toHaveProperty('tx_id', transactions.tx40.hash);
+    expect(tkaUtxos.utxos[2]).toHaveProperty('tx_id', transactions.tx50.hash);
+    expect(tkaUtxos.utxos[3]).toHaveProperty('tx_id', transactions.tx50.hash);
+
+    /*
+     * Tx50 was the last executed transaction, so it has two outputs: 50 tka for one address
+     * and the remaining change for address 0. The order that the UTXOs from tx50 appear are not
+     * consistent, so we make a simple check to solve this.
+     */
+    const tkaChangeOnAddr0 = 1000 - 10 - 20 - 30 - 40 - 50;
+    if (tkaUtxos.utxos[2].amount === 50) {
+      expect(tkaUtxos.utxos[3]).toHaveProperty('amount', tkaChangeOnAddr0);
+    } else {
+      expect(tkaUtxos.utxos[2]).toHaveProperty('amount', tkaChangeOnAddr0);
+      expect(tkaUtxos.utxos[3]).toHaveProperty('amount', 50);
+    }
+  });
+
+  it('should return correct results for maximum_amount', async () => {
+    const utxoResponse = await TestUtils.request
+      .get('/wallet/utxo-filter')
+      .query({
+        token: tokenA.uid,
+        maximum_amount: 100
+      })
+      .set({ 'x-wallet-id': wallet2.walletId });
+    const tkaUtxos = utxoResponse.body;
+
+    expect(tkaUtxos.total_amount_available).toBe(100);
+    expect(tkaUtxos.total_utxos_available).toBe(4);
+    expect(tkaUtxos.utxos).toHaveProperty('length', 4);
+
+    // We expect these UTXOs to be in the same order as a normal query.
+    expect(tkaUtxos.utxos[0]).toHaveProperty('tx_id', transactions.tx10.hash);
+    expect(tkaUtxos.utxos[1]).toHaveProperty('tx_id', transactions.tx20.hash);
+    expect(tkaUtxos.utxos[2]).toHaveProperty('tx_id', transactions.tx30.hash);
+    expect(tkaUtxos.utxos[3]).toHaveProperty('tx_id', transactions.tx40.hash);
+  });
+
+  it('should return correct results for maximum_amount and bigger_than', async () => {
+    const utxoResponse = await TestUtils.request
+      .get('/wallet/utxo-filter')
+      .query({
+        token: tokenA.uid,
+        maximum_amount: 100,
+        amount_bigger_than: 30
+      })
+      .set({ 'x-wallet-id': wallet2.walletId });
+    const tkaUtxos = utxoResponse.body;
+
+    expect(tkaUtxos.total_amount_available).toBe(70);
+    expect(tkaUtxos.total_utxos_available).toBe(2);
+    expect(tkaUtxos.utxos).toHaveProperty('length', 2);
+
+    expect(tkaUtxos.utxos[0]).toHaveProperty('tx_id', transactions.tx30.hash);
+    expect(tkaUtxos.utxos[1]).toHaveProperty('tx_id', transactions.tx40.hash);
+  });
+
+  // It seems there is a name mismatch on the lib parameter: only_available => only_available_utxos
```

Comment:
The API does not work with this parameter? Do we have an issue for that? I think we should add this to the KTLO spreadsheet, it's an important issue.

Reply:
This is mapped as one of the tasks on issue 161 for fixing the skipped tests.

I can add a KTLO for that as well :+1:

Reply:
Yes, I think you should create a specific issue for that and add it to the KTLO.

Reply:
Created #179 for fixing this issue